okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -577,13 +606,27 @@ sslSocketFactory, hostnameVerifier, sock, getOverridenHost(), getOverridenPort()
           clientFrameHandler = new ClientFrameHandler(variant.newReader(source, true));
         }
         synchronized (lock) {
+          latchForExtraThread.countDown();
           socket = Preconditions.checkNotNull(sock, "socket");
           if (sslSession != null) {
             securityInfo = new InternalChannelz.Security(new InternalChannelz.Tls(sslSession));
           }
         }
       }
     });
+
+    executor.execute(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          System.out.println("In other thread");
+          barrier.await();
```

Comment:
We will want a time limit here as well, as runnables aren't guaranteed to run in the enqueued order.